### DISCUSSION TITLE
understanding the audit error (cuda/rocm variants)

### DISCUSSION THREAD

**Discussion by Thomas-Ulrich** [spack/spack#42786] [Q&A] [answered]
in https://github.com/spack/spack/pull/42785, the audit is failing, and I don't really understand how to fix it.

```
Run . share/spack/setup-env.sh
PKG-DIRECTIVES: 2 issues found
1. tandem: wrong variant in "depends_on" directive
    the variant 'cuda' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/tandem/package.py
2. tandem: wrong variant in "depends_on" directive
    the variant 'rocm' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/tandem/package.py
Error: Process completed with exit code 1.
```

Should I add 2 variants cuda and rocm to tandem?

**Comment by alalazo** [comment]
The PR is adding directives in `tandem` like:
```python
depends_on("mpich+cuda", when="+cuda ^[virtuals=mpi] mpich")
```
If the `cuda` variant isn't defined on `tandem`, then the `when` directive is always false. The audit is pointing that out.

Does the package currently behave as you expect during concretization?

**Reply by Thomas-Ulrich** [reply, marked as answer]
Thanks, I forgot to derive tandem of CudaPackage and  ROCmPackage, now it is fixed.
